src/server/session.ts

```diff
@@ -2121,26 +2211,73 @@ namespace ts.server {
                 );
         }
 
-        private getFullNavigateToItems(args: protocol.NavtoRequestArgs): CombineOutputResult<NavigateToItem> {
+        private getFullNavigateToItems(args: protocol.NavtoRequestArgs): ProjectNavigateToItems[] {

```

Comment:
This no longer uses a shared helper and there are behavioral changes (see baseline).  I also attempted to get rid of some of the linear searching, but JS wasn't very cooperative.